src/EditorFeatures/CSharp/QuickInfo/SyntacticQuickInfoProvider.cs

```diff
@@ -76,8 +85,66 @@ internal class SyntacticQuickInfoProvider : AbstractQuickInfoProvider
                 return null;
             }
 
-            var span = new SnapshotSpan(textSnapshot, Span.FromBounds(parent.SpanStart, openBrace.Span.End));
+            var span = new SnapshotSpan(textSnapshot, Span.FromBounds(spanStart, spanEnd));
             return this.CreateElisionBufferDeferredContent(span);
         }
+
+        private static void MarkInterestedSpanNearbyScopeBlock(SyntaxNode block, SyntaxToken openBrace, ref int spanStart, ref int spanEnd)
+        {
+            SyntaxTrivia nearbyTrivia;
+
+            if (openBrace.HasLeadingTrivia && FindFurthestNearbyComment(openBrace.LeadingTrivia.Reverse(), out nearbyTrivia))
+            {
+                spanStart = nearbyTrivia.SpanStart;
+                return;
+            }
+
+            var nextToken = block.FindToken(openBrace.FullSpan.End);
+            if (nextToken.HasLeadingTrivia && FindFurthestNearbyComment(nextToken.LeadingTrivia, out nearbyTrivia))
+            {
+                spanEnd = nearbyTrivia.Span.End;
+                return;
+            }
+        }
+
+        private static bool FindFurthestNearbyComment(IEnumerable<SyntaxTrivia> triviaSearchList, out SyntaxTrivia nearbyTrivia)
```

Comment:
@CyrusNajmabadi My current implementation only allow one kind of comments that immediately above/below the open brace.

Assuming you allow
```C#
/* haha */

// dada

{
```
Then I am OK, we can even loosen the check to first non-empty trivia so that this method is not needed, so, your call?